src/lib.rs

```diff
@@ -151,190 +152,34 @@ pub trait PolynomialCommitment<F: Field>: Sized {
-    fn open_combinations<'a>(
-        ck: &Self::CommitterKey,
-        linear_combinations: impl IntoIterator<Item = &'a LinearCombination<F>>,
-        polynomials: impl IntoIterator<Item = &'a LabeledPolynomial<F>>,
-        commitments: impl IntoIterator<Item = &'a LabeledCommitment<Self::Commitment>>,
-        query_set: &QuerySet<F>,
-        fs_rng: &mut Self::RandomOracle,
-        rands: impl IntoIterator<Item = &'a LabeledRandomness<Self::Randomness>>,
-        rng: Option<&mut dyn RngCore>,
-    ) -> Result<BatchLCProof<F, Self>, Self::Error>
-        where
-            Self::Randomness: 'a,
             Self::Commitment: 'a,
-    {
-        Self::open_combinations_individual_opening_challenges(
-            ck,
-            linear_combinations,
-            polynomials,
-            commitments,
-            query_set,
-            fs_rng,
-            rands,
-            rng,
-        )
-    }
+            Self::Randomness: 'a;
 
-    /// Multi point multi LC verify.
-    /// Checks that `evaluations` are the true evaluations at `query_set` of the
-    /// linear combinations of polynomials committed in `commitments`.
-    fn check_combinations<'a>(
-        vk: &Self::VerifierKey,
-        linear_combinations: impl IntoIterator<Item = &'a LinearCombination<F>>,
-        commitments: impl IntoIterator<Item = &'a LabeledCommitment<Self::Commitment>>,
-        eqn_query_set: &QuerySet<F>,
-        eqn_evaluations: &Evaluations<F>,
-        proof: &BatchLCProof<F, Self>,
-        fs_rng: &mut Self::RandomOracle,
-    ) -> Result<bool, Self::Error>
-        where
-            Self::Commitment: 'a,
-    {
-        Self::check_combinations_individual_opening_challenges(
-            vk,
-            linear_combinations,
-            commitments,
-            eqn_query_set,
-            eqn_evaluations,
-            proof,
-            fs_rng,
-        )
-    }
 
     /// Single point multi poly open, allowing the random oracle to be passed from
     /// 'outside' to the function.
     /// CAUTION: This is a low-level function to be handled carefully, typically
     /// presuming that commitments and query_set is already bound to the internal
     /// state of the Fiat-Shamir rng.
-    /// TODO: rename this function
-    fn open_individual_opening_challenges<'a>(
+    fn single_point_multi_poly_open<'a>(
         ck: &Self::CommitterKey,
-        labeled_polynomials: impl IntoIterator<Item = &'a LabeledPolynomial<F>>,
+        labeled_polynomials: impl IntoIterator<Item = &'a LabeledPolynomial<G::ScalarField>>,
         commitments: impl IntoIterator<Item = &'a LabeledCommitment<Self::Commitment>>,
-        point: F,
+        point: G::ScalarField,
         fs_rng: &mut Self::RandomOracle,
         rands: impl IntoIterator<Item = &'a LabeledRandomness<Self::Randomness>>,
-        // `rng` if needed for blinding 
+        // `rng` if needed for blinding
         rng: Option<&mut dyn RngCore>,
     ) -> Result<Self::Proof, Self::Error>
```

Comment:
We decided to put the implementation for this function in the trait. 
As it is now the trait of a linear polynomial commitment scheme, that is the right place.

Reply:
Done